modules/nux/modules.php

```diff
@@ -245,6 +245,293 @@ public function process() {
+             } else {
+                 /*
+                 *  Connect to SMTP server if user wants to send emails
+                 */
+                 if($form['nux_config_is_sender']){
+                     $result = $this->connectToSMPT(
+                        $form['nux_config_smtp_address'],
+                        $form['nux_config_profile_name'],
+                        $form['nux_config_smtp_port'],
+                        $form['nux_config_email'],
+                        $form['nux_config_password'],
+                        $form['nux_config_smtp_tls']);
+    
+                     if(!isset($result)) return;
+                 }
+    
+                 /*
+                  *  Connect to IMAP server if user wants to receive emails
+                  */
+                 if($form['nux_config_is_receiver']){
+                     $result = $this->connectToIMAP(
+                       $form['nux_config_imap_address'],
+                       $form['nux_config_profile_name'],
+                       $form['nux_config_imap_port'],
+                       $form['nux_config_email'],
+                       $form['nux_config_password'],
+                       $form['nux_config_imap_tls'],
+                       $form['nux_imap_sieve_host'],
+                       $form['nux_enable_sieve']);
+    
+                     if(!isset($result)) {
+                       /**
+                       * Delete the SMPT server created before
+                       **/
+                       if(!isset($result) && $form['nux_config_is_sender']){
+                           $res = Hm_SMTP_List::del($this->smtp_server_id);
+                           if ($res) {
+                               Hm_SMTP_List::forget_credentials($this->smtp_server_id);
+                               $smtp_servers = Hm_SMTP_List::dump(false, true);
+                               $this->user_config->set('smtp_servers', $smtp_servers);
+                           }
+                       }
+                        return;
+                     }
+                 }
+    
+                 if($form['nux_config_is_sender'] && $form['nux_config_is_receiver'] && $form['nux_create_profile'] && isset($this->imap_server_id) && isset($this->smtp_server_id)) {
+                     $this->saveProfile(
+                        $form['nux_config_profile_name'],
+                        $form['nux_profile_signature'],
+                        $form['nux_profile_reply_to'],
+                        $form['nux_profile_is_default'],
+                        $form['nux_config_email'],
+                        $form['nux_config_imap_address']
+                        );
+                 }
+    
+                 Hm_Msgs::add("Server saved");
+                 $this->out('just_saved_credentials', $this->just_saved_credentials);
+            }
+        }
+    }
+
+    public function saveProfile($name, $signature, $replyTo, $isDefault, $email, $serverMail){
```

Comment:
Cypht module files contain classes that are responsible for request handling (handlers) and output. Please move these auxiliary methods to a separate class or set of classes/function files or whatever makes sense. It will also make the modules file shorter and easier to follow. E.g. connecting to an IMAP server makes sense to live in the imap module. Saving a profile should probably go in the profile module.